packages/deployer/task-names.js

```diff
@@ -1,18 +1,18 @@
 module.exports = {
-  SUBTASK_CANCEL_DEPLOYMENT: 'cancel-deployment',
-  SUBTASK_CLEAR_DEPLOYMENTS: 'clear-deployments',
-  SUBTASK_DEPLOY_CONTRACT: 'deploy-contract',
-  SUBTASK_DEPLOY_CONTRACTS: 'deploy-contracts',
-  SUBTASK_DEPLOY_MODULES: 'generate-deploy-modules',
-  SUBTASK_DEPLOY_ROUTER: 'generate-deploy-router',
-  SUBTASK_GENERATE_ROUTER_SOURCE: 'generate-router-source',
-  SUBTASK_PREPARE_DEPLOYMENT: 'prepare-deployment',
-  SUBTASK_PRINT_INFO: 'print-info',
-  SUBTASK_SYNC_PROXY: 'sync-proxy',
-  SUBTASK_SYNC_SOURCES: 'sync-sources',
-  SUBTASK_UPGRADE_PROXY: 'upgrade-proxy',
-  SUBTASK_VALIDATE_MODULES: 'validate-modules',
-  SUBTASK_VALIDATE_STORAGE: 'validate-storage',
-  SUBTASK_VALIDATE_ROUTER: 'validate-router',
+  SUBTASK_CANCEL_DEPLOYMENT: 'deployer-cancel-deployment',
```

Comment:
Had to prepend task names with the plugin, to avoid collisions. Looks like hardhat mixes all plugin subtasks in the same environment.